torch/nn/modules/conv.py

```diff
@@ -173,17 +176,23 @@ class Conv1d(_ConvNd):
     """
 
     def __init__(self, in_channels, out_channels, kernel_size, stride=1,
-                 padding=0, dilation=1, groups=1, bias=True):
+                 padding=0, dilation=1, groups=1,
+                 bias=True, padding_mode='zeros'):
         kernel_size = _single(kernel_size)
         stride = _single(stride)
         padding = _single(padding)
         dilation = _single(dilation)
         super(Conv1d, self).__init__(
             in_channels, out_channels, kernel_size, stride, padding, dilation,
-            False, _single(0), groups, bias)
+            False, _single(0), groups, bias, padding_mode)
 
     @weak_script_method
     def forward(self, input):
+        if self.padding_mode == 'circular':
+            expanded_padding = ((self.padding[0] + 1) // 2, self.padding[0] // 2)
+            return F.conv1d(F.pad(input, expanded_padding, mode='circular'),
+                            self.weight, self.bias, self.stride,
+                            _single(0), self.dilation, self.groups)
         return F.conv1d(input, self.weight, self.bias, self.stride,
```

Comment:
do we want to have the functional interface of `conv` also support the padding mode?

Reply:
In the documentation it appears padding_mode is supported, but when I try to use it tells me it is an unexpected argument. I will do the manual padding, as in the module, but it would be convenient if the operation is supported.
In my case, my filter is a tensor, not a parameter, and I would prefer not to use a module.

Thanks!